### PR TITLE
docs(upgrade): Documentation for ng-metadata/upgrade

### PR DESCRIPTION
**Pull request by JamesHenry**

### CODE REVIEW

File: SUMMARY.md [hunk 1]
```diff
@@ -33,6 +34,8 @@
     * [ng-metadata/router-deprecated](/docs/api/router-deprecated/README.md)
       * [Class](/docs/api/router-deprecated/class.md)
       * [Interface](/docs/api/router-deprecated/interface.md)
+    * [ng-metadata/upgrade](/docs/api/upgrade/README.md)
+      * [Function](/docs/api/upgrade/function.md)
```
Comment (Hotell) [inline SUMMARY.md]: this is a `Class` category

File: docs/api/upgrade/function.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+## NgMetadataUpgradeAdapter
+
+Please see the [ngUpgrade recipe](/docs/recipes/ng-upgrade.md) for how to use the NgMetadataUpgradeAdapter.
+
+###### Parameters
+
+| Parameter           | Type                           | Description                                               |
+| --------------------| -------------------------------|-----------------------------------------------------------|
+| **UpgradeAdapter**  | FunctionConstructor            | Official UpgradeAdapter constructor from @angular/upgrade |
```
Comment (Hotell) [inline docs/api/upgrade/function.md]: rather `Type` than `FunctionConstructor`, Typescript like

File: SUMMARY.md [hunk 3]
```diff
@@ -15,6 +15,7 @@
     * [UI Router](docs/recipes/ui-router.md)
     * [Deprecated Router](docs/recipes/deprecated-router.md)
     * [Migrating from ngMetadata 1.x to 2.0](docs/recipes/migration-from-ng-metadata-1.x.md)
+    * [Hybrid apps with ngUpgrade](docs/recipes/ng-upgrade.md)
```
Comment (Hotell) [inline SUMMARY.md]: maybe `Hybrid ng2 and ng1 apps with ngUpgrade` sounds more explicit ? WDYT?

File: docs/api/upgrade/function.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Function
+
+- [NgMetadataUpgradeAdapter](#NgMetadataUpgradeAdapter)
+
+---
+
+## NgMetadataUpgradeAdapter
```
Comment (Hotell) [inline docs/api/upgrade/function.md]: I would also describe all method available on instance of `NgMetadataUpgradeAdapter`. user will see them immediately thanks to TS, but I would like to have them in docs to be consistent

**Comment by Hotell** [comment]
Review done, just few minor tweaks and we can merge.

Overall, the docs are pure gold! Well done!

**Comment by JamesHenry** [comment]
Updated, PTAL

**Pull request merged**